### PR TITLE
[Neural Speed] Reload if `ctx_size` changes and add `reset_kv_cache` API

### PR DESCRIPTION
**Pull request by zhentaoyu**
## Type of Change

feature or bug fix or documentation or others
API changed or not

## Description

detail description 
Issues: xxx

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Review by zhenwei-intel — APPROVED** [review]

**Review by a32543254 — APPROVED** [review]
LGTM

File: neural_speed/__init__.py [hunk 1]
```diff
@@ -402,6 +403,12 @@ def generate(self,
     def is_token_end(self):
         return self.model.is_token_end()
 
+    def reset_kv_cache(self):
+        if self.model is None:
+            return
```
Comment (a32543254) [inline neural_speed/__init__.py:408]: if self.model is None:
  may be we could throw a warning log like:
  The model is not inited yet, then return

File: neural_speed/__init__.py [hunk 2]
```diff
@@ -434,6 +441,7 @@ def __call__(self, model_input, reinit=False, logits_all=False, **kwargs):
             logits_seq_len_dim = model_input.shape[1] if logits_all else 1
             self.reinit_from_bin = False
             self._check_max_request_num(batch_size, **kwargs)
+            self._check_ctx_size(kwargs.get("ctx_size", 0), multi_round=(not reinit))
```
Comment (a32543254) [inline neural_speed/__init__.py:444]: multi_round=(not reinit)   ?

should be bool right?

File: neural_speed/models/llama/llama_utils.cpp [hunk 3]
```diff
@@ -65,7 +65,7 @@ void Llama::init(const char* path_model, model_context* ctx, int n_gpu_layer_, b
   auto& hparams = model.hparams;
   n_ff = hparams.ffn_hidden_size;
   fprintf(stderr, "%s: n_vocab    = %u\n", __func__, hparams.n_vocab);
-  fprintf(stderr, "%s: n_ctx      = %u\n", __func__, hparams.max_seq_len);
+  fprintf(stderr, "%s: n_ctx      = %u\n", __func__, lctx.n_ctx);
```
Comment (a32543254) [inline neural_speed/models/llama/llama_utils.cpp:68]: what is the default value of  lctx.n_ctx ?

File: neural_speed/__init__.py (hunk 2 above)
Comment (zhentaoyu) [reply neural_speed/__init__.py:444]: it is bool

File: neural_speed/models/llama/llama_utils.cpp (hunk 3 above)
Comment (zhentaoyu) [reply neural_speed/models/llama/llama_utils.cpp:68]: https://github.com/intel/neural-speed/blob/main/neural_speed/models/model_utils/model_types.h#L307

File: neural_speed/__init__.py (hunk 1 above)
Comment (zhentaoyu) [reply neural_speed/__init__.py:409]: fixed in [47c3f09](https://github.com/intel/neural-speed/pull/227/commits/47c3f0967eeaff42b6b31286f03dcba3af721743)

**Comment by zhentaoyu** [comment]
> Could you please clarify n_ctx, n_tokens, n_total?
> 
> seems there is no the lengths of input tokens?

- `n_ctx` is the context window, which stands for the `seq_length` dim of `kv_cache` in memory.
- `n_tokens` is the input prompt tokens length. You can see them in each model cpp eval func.
- `n_total` is the all processed tokens in generation history which was added in the `streaming-llm` feature. It is always `>= n_past` and you do not need to care about it if your codes have no related to  the `streaming-llm`.
